ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -187,28 +190,79 @@ internal constructor(
     externalScope.launch(ioDispatcher) { submitDataUseCase.invoke(loiId, job, surveyId, deltas) }
   }
 
-  /** Returns the position of the task fragment visible to the user. */
-  fun getVisibleTaskPosition() = currentPosition.value
-
-  /** Displays the task at the given position to the user. */
-  fun updateCurrentPosition(position: Int) {
-    savedStateHandle[TASK_POSITION_KEY] = position
+  fun getAbsolutePosition(): Int {
+    if (currentTaskId.value == "") {
+      return 0
+    }
+    return tasks.indexOf(tasks.first { it.id == currentTaskId.value })
   }
 
-  /** Returns true if the given task position is last. */
-  fun isLastPosition(taskPosition: Int): Boolean {
-    val finalTaskPosition = tasks.size - 1
+  /** Get the index of the current task within the (possibly conditional) task sequence. */
+  fun getRelativePosition(): Pair<Int, Int> {
+    var currentIndex = 0
+    var size = 0
+    getTaskSequence().forEachIndexed { index, task ->
+      if (task.id == currentTaskId.value) {
+        currentIndex = index
+      }
+      size++
+    }
+    return currentIndex to size
+  }
 
-    assert(finalTaskPosition >= 0)
-    assert(taskPosition in 0..finalTaskPosition)
+  /**
+   * Retrieves the current task sequence given the inputs and conditions set on the tasks. Setting a
+   * start ID will always generate a sequence with the start ID as the first element, and if
+   * preceding is set, will generate the previous tasks from there.
+   */
+  private fun getTaskSequence(startId: String? = null, preceding: Boolean = false): Sequence<Task> {
```

Comment:
Isn't "preceding" really "reversed"? Not clear on the abstraction here, I'll ask now on VC.